src/clojure_mauth_client/middleware.clj

```diff
@@ -20,9 +20,20 @@
                                 :else (slurp body))
           headers (-> (:headers request)
                       downcase-header-keys)
-          [mauth-time mauth-auth mauth-version] (if (every? headers ["mcc-time" "mcc-authentication"])
-                                                  [(get headers "mcc-time") (get headers "mcc-authentication") "v2"]
-                                                  [(get headers "x-mws-time") (get headers "x-mws-authentication") "v1"])
+          [mauth-time mauth-auth mauth-version] (cond
+                                                  (every? headers
+                                                          ["mcc-time"
+                                                           "mcc-authentication"])     [(get headers "mcc-time")
+                                                                                       (get headers "mcc-authentication")
+                                                                                       "v2"]
```

Comment:
Please refer to the MAuth specification and make sure to interpret the headers in the way the spec describes. Just because we have the `mcc-authentication` header, that doesn't mean it's V2. We need to look for the `"MWSV2"` token.

Reply:
@colinphill-mdsol  - That validation is done in validate.clj

Reply:
Hmm, okay, I see that now. It still doesn't seem quite correct; we shouldn't assume that it's V2 in the first place. Instead, we should use that token to determine the version.

But since we've agreed that this is all temporary and we're going to reimplement it later, this feedback is non-blocking.

Reply:
@colinphill-mdsol - Done suggested changes.